src/main/java/com/example/nlern/models/Mark.java

```diff
@@ -1,85 +1,39 @@
 package com.example.nlern.models;
 
 
-import jakarta.persistence.*;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import jakarta.persistence.Id;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.OneToOne;
+
+import lombok.Getter;
+import lombok.Setter;
 import org.hibernate.annotations.OnDelete;
 import org.hibernate.annotations.OnDeleteAction;
 
-import java.util.List;
-
 @Entity
 @Table(name = "marks")
+@Getter @Setter
 public class Mark {
-
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private int id;
 
-
     private int mark;
-
     private int scale;
 
-
-
     @ManyToOne
     @OnDelete(action = OnDeleteAction.CASCADE)
```

Comment:
You are mixing different approaches of cascade delete
Either use `@OnDelete` everywhere or use `cascade` of `@OneToOne`/`@OneToMany` annotations

Reply:
I use uniderectional mapping, and in some situations I can only use `@OnDelete`, and in other cases only `@OnDelete.`